src/server/stream_family.cc

```diff
@@ -636,6 +686,54 @@ OpResult<RecordVec> OpRange(const OpArgs& op_args, string_view key, const RangeO
   return result;
 }
 
+OpResult<RecordVec> OpRangeFromConsumerPEL(const OpArgs& op_args, string_view key,
+                                           const RangeOpts& opts) {
+  RecordVec result;
+
+  if (opts.count == 0)
+    return result;
+
+  unsigned char start_key[sizeof(streamID)];
+  unsigned char end_key[sizeof(streamID)];
+  auto sstart = opts.start.val;
+  auto send = opts.end.val;
+
+  StreamEncodeID(start_key, &sstart);
+  StreamEncodeID(end_key, &send);
+  raxIterator ri;
+
+  raxStart(&ri, opts.consumer->pel);
+  raxSeek(&ri, ">=", start_key, sizeof(start_key));
+  size_t ecount = 0;
+  while (raxNext(&ri) && (!opts.count || ecount < opts.count)) {
+    if (memcmp(ri.key, &send, ri.key_len) > 0)
+      break;
+    streamID id;
+
+    streamDecodeID(ri.key, &id);
+    RangeOpts ropts;
+    ropts.start.val = id;
+    ropts.end.val = id;
+    ropts.group = nullptr;
+    ropts.consumer = nullptr;
```

Comment:
Too verbose, consider adding a constructor (I would suggest the non standard before C++ 20 https://en.cppreference.com/w/cpp/language/aggregate_initialization#Designated_initializers but I really don't like non conforming C++ standard code -- well with a few exceptions ;))